src/newgrf_object.cpp

```diff
@@ -87,37 +101,45 @@ bool ObjectSpec::IsAvailable() const
  */
 uint ObjectSpec::Index() const
 {
-	return this - _object_specs;
+	return this - &_object_specs[0];
```

Comment:
```suggestion
	return this - _object_specs.data();
```
I think it's better to use `data()`